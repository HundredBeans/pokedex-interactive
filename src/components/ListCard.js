import { Box, Flex, Heading, Image, Badge } from "@chakra-ui/react";
import React from "react";
import placeholder from "../assets/placeholder.png";

const ListCard = (props) => {
  return (
    <Box
      borderRadius="lg"
      _hover={{ boxShadow: "lg" }}
      px={2}
      boxShadow="xs"
      onClick={props.onClickCard}
    >
      <Flex flexDirection="column" justify="center" align="center" py={5}>
        <Heading as="h3" size="md">
          {props.name.toString().toUpperCase()}
        </Heading>
        <Image fallbackSrc={placeholder} boxSize="130px" src={props.imageSrc} />
        <Badge variant="subtle" colorScheme={props.owned ? "green" : "red"}>
          Owned - {props.owned || 0}{" "}
        </Badge>
      </Flex>
    </Box>
  );
};

export default ListCard;
