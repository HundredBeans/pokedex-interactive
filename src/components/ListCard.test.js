import { cleanup } from "@testing-library/react";
import React from "react";
import { render } from "../test-utils";
import ListCard from "./ListCard";

describe("ListCard.js test", () => {
  let renderedComponent;

  const onClickCardMock = jest.fn();
  const pokemonData = {
    name: "Bulbasaur",
    image:
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png",
    owned: 1,
  };

  beforeEach(() => {
    renderedComponent = render(
      <ListCard
        name={pokemonData.name}
        imageSrc={pokemonData.image}
        owned={pokemonData.owned}
        onClickCard={onClickCardMock}
        isLoaded={true}
      />
    );
  });

  afterEach(() => {
    cleanup();
  });

  it("renders the component (loaded)", () => {
    expect(renderedComponent).toMatchSnapshot("loaded");
    // Check if the card is showing / visible
    const nameElement = renderedComponent.getByText(
      pokemonData.name.toString().toUpperCase()
    );
    expect(nameElement).toBeVisible();
  });

  it("renders the component (loaded without owned)", () => {
    renderedComponent.rerender(
      <ListCard
        name={pokemonData.name}
        imageSrc={pokemonData.image}
        onClickCard={onClickCardMock}
        isLoaded={false}
      />
    );
    expect(renderedComponent).toMatchSnapshot("loaded-without-owned");
  });

  it("renders the component (loading)", () => {
    renderedComponent.rerender(
      <ListCard
        name={pokemonData.name}
        imageSrc={pokemonData.image}
        owned={pokemonData.owned}
        onClickCard={onClickCardMock}
        isLoaded={false}
      />
    );
    expect(renderedComponent).toMatchSnapshot("loading");
  });
});
