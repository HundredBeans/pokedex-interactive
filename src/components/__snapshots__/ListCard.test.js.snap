// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListCard.js test renders the component (loaded without owned): loaded-without-owned 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto; top: 0px; right: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; top: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; top: 0px; right: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; bottom: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto; bottom: 0px; right: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; bottom: 0px; right: 0px;"
      />
    </div>
    <div>
      <div
        class="css-dglzss"
      >
        <div
          class="css-13i2prf"
        >
          <div
            class="chakra-skeleton css-40mwtr"
          >
            <h3
              class="chakra-heading css-ns7u32"
            >
              BULBASAUR
            </h3>
          </div>
          <div
            class="chakra-skeleton css-1mgb0mg"
          >
            <img
              class="chakra-image__placeholder css-1t37noe"
              src="placeholder.png"
            />
          </div>
          <div
            class="chakra-skeleton css-40mwtr"
          >
            <span
              class="chakra-badge css-1b8y92l"
            >
              Owned - 
              0
               
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-dglzss"
    >
      <div
        class="css-13i2prf"
      >
        <div
          class="chakra-skeleton css-40mwtr"
        >
          <h3
            class="chakra-heading css-ns7u32"
          >
            BULBASAUR
          </h3>
        </div>
        <div
          class="chakra-skeleton css-1mgb0mg"
        >
          <img
            class="chakra-image__placeholder css-1t37noe"
            src="placeholder.png"
          />
        </div>
        <div
          class="chakra-skeleton css-40mwtr"
        >
          <span
            class="chakra-badge css-1b8y92l"
          >
            Owned - 
            0
             
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ListCard.js test renders the component (loaded): loaded 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto; top: 0px; right: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; top: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; top: 0px; right: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; bottom: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto; bottom: 0px; right: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; bottom: 0px; right: 0px;"
      />
    </div>
    <div>
      <div
        class="css-dglzss"
      >
        <div
          class="css-13i2prf"
        >
          <div
            class="chakra-skeleton css-p93ydk"
          >
            <h3
              class="chakra-heading css-ns7u32"
            >
              BULBASAUR
            </h3>
          </div>
          <div
            class="chakra-skeleton css-cdkrf0"
          >
            <img
              class="chakra-image__placeholder css-1t37noe"
              src="placeholder.png"
            />
          </div>
          <div
            class="chakra-skeleton css-p93ydk"
          >
            <span
              class="chakra-badge css-eywthe"
            >
              Owned - 
              1
               
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-dglzss"
    >
      <div
        class="css-13i2prf"
      >
        <div
          class="chakra-skeleton css-p93ydk"
        >
          <h3
            class="chakra-heading css-ns7u32"
          >
            BULBASAUR
          </h3>
        </div>
        <div
          class="chakra-skeleton css-cdkrf0"
        >
          <img
            class="chakra-image__placeholder css-1t37noe"
            src="placeholder.png"
          />
        </div>
        <div
          class="chakra-skeleton css-p93ydk"
        >
          <span
            class="chakra-badge css-eywthe"
          >
            Owned - 
            1
             
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ListCard.js test renders the component (loading): loading 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto; top: 0px; right: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; top: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; top: 0px; right: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; bottom: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto; bottom: 0px; right: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; bottom: 0px; right: 0px;"
      />
    </div>
    <div>
      <div
        class="css-dglzss"
      >
        <div
          class="css-13i2prf"
        >
          <div
            class="chakra-skeleton css-40mwtr"
          >
            <h3
              class="chakra-heading css-ns7u32"
            >
              BULBASAUR
            </h3>
          </div>
          <div
            class="chakra-skeleton css-1mgb0mg"
          >
            <img
              class="chakra-image__placeholder css-1t37noe"
              src="placeholder.png"
            />
          </div>
          <div
            class="chakra-skeleton css-40mwtr"
          >
            <span
              class="chakra-badge css-eywthe"
            >
              Owned - 
              1
               
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-dglzss"
    >
      <div
        class="css-13i2prf"
      >
        <div
          class="chakra-skeleton css-40mwtr"
        >
          <h3
            class="chakra-heading css-ns7u32"
          >
            BULBASAUR
          </h3>
        </div>
        <div
          class="chakra-skeleton css-1mgb0mg"
        >
          <img
            class="chakra-image__placeholder css-1t37noe"
            src="placeholder.png"
          />
        </div>
        <div
          class="chakra-skeleton css-40mwtr"
        >
          <span
            class="chakra-badge css-eywthe"
          >
            Owned - 
            1
             
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
